### DISCUSSION TITLE
Ability to use any custom tflite model

### DISCUSSION THREAD

**Discussion by abhinavchawla13** [capawesome-team/capacitor-mlkit#103] [Ideas]
It would be nice to have a wrapper around MLkit where we can pass in the input tensor from the Capacitor into this generic plugin which can be setup with a custom tflite file. Optionally, we can have the option for pre-processing and/or post-processing within the plugin.

At least the ability to pre-process (convert image to tensor, scale the tensor, and so on) can be done on the Capacitor side (for instance, using Tensorflow JS) and then passing it through the plugin for inference using hardware GPU using MLKit, and returning the tensors (or converted array object) back would be super helpful to start using in a more generic way. 

Would love to know any constraints or challenges associated with it!

**Comment by robingenz** [comment]
As i said in https://github.com/capawesome-team/capacitor-mlkit/discussions/102, for Tensorflow, a custom Capacitor plugin with a specific Tensorflow implementation would be necessary.

> At least the ability to pre-process (convert image to tensor, scale the tensor, and so on) can be done on the Capacitor side (for instance, using Tensorflow JS) and then passing it through the plugin for inference using hardware GPU using MLKit, and returning the tensors (or converted array object) back would be super helpful to start using in a more generic way.

Which ML Kit API accepts tensors and returns tensors? As far as I can see, the APIs are all high-level and you pass the images directly.
